unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/EmotesCustomization/EmoteCardComponentView.cs

```diff
@@ -0,0 +1,372 @@
+
+        /// <summary>
+        /// Set the emote as selected.
+        /// </summary>
+        /// <param name="isSelected">True for selecting it.</param>
+        void SetEmoteAsSelected(bool isSelected);
+
+        /// <summary>
+        /// Set the type of rarity in the card.
+        /// </summary>
+        /// <param name="rarity">New rarity.</param>
+        void SetRarity(string rarity);
+
+        /// <summary>
+        /// Set the emote as L2 or not.
+        /// </summary>
+        /// <param name="isInL2">True for set it as L2.</param>
+        void SetIsInL2(bool isInL2);
+
+        /// <summary>
+        /// Set the emote as collectible or not.
+        /// </summary>
+        /// <param name="isInL2">True for set it as collectible.</param>
+        void SetIsCollectible(bool isCollectible);
+
+        /// <summary>
+        /// Set the emote card as loading state or not.
+        /// </summary>
+        /// <param name="isLoading">True for setting it as loading state.</param>
+        void SetAsLoading(bool isLoading);
+    }
+
+    public class EmoteCardComponentView : BaseComponentView, IEmoteCardComponentView, IComponentModelConfig
+    {
+        internal static readonly int ON_SELECTED_CARD_COMPONENT_BOOL = Animator.StringToHash("OnSelected");
+
+        [Header("Prefab References")]
+        [SerializeField] internal ImageComponentView emoteImage;
+        [SerializeField] internal GameObject emoteNameContainer;
+        [SerializeField] internal TMP_Text emoteNameText;
+        [SerializeField] internal TMP_Text assignedSlotNumberText;
+        [SerializeField] internal ButtonComponentView mainButton;
+        [SerializeField] internal ButtonComponentView infoButton;
+        [SerializeField] internal GameObject cardSelectionFrame;
+        [SerializeField] internal Animator selectionAnimator;
+        [SerializeField] internal Image rarityMark;
+        [SerializeField] internal Transform emoteInfoAnchor;
+        [SerializeField] internal GameObject loadingSpinnerGO;
+
+        [Header("Configuration")]
+        [SerializeField] internal Sprite defaultEmotePicture;
+        [SerializeField] internal Sprite nonEmoteAssignedPicture;
+        [SerializeField] internal List<EmoteRarity> rarityColors;
+        [SerializeField] internal EmoteCardComponentModel model;
+
+        public Button.ButtonClickedEvent onMainClick => mainButton?.onClick;
+        public Button.ButtonClickedEvent onInfoClick => infoButton?.onClick;
+        
+        public event Action<string> onEmoteSelected;
+
+        public override void Awake()
+        {
+            base.Awake();
+
+            if (emoteImage != null)
+                emoteImage.OnLoaded += OnEmoteImageLoaded;
+
+            if (cardSelectionFrame != null)
+                cardSelectionFrame.SetActive(false);
+        }
+
+        public void Configure(BaseComponentModel newModel)
+        {
+            model = (EmoteCardComponentModel)newModel;
```

Comment:
In all these different View scripts, can it happen that the `Configure()` is called again with the same model as before? if that's the case we should check here if the `newModel` is actually different than the old one before refreshing the whole UI. Otherwise it's not needed!

Reply:
In the current implementations of this knd of views shouldn't be happening that because the own consumers are taking care of avoiding multiples calls, but yes, if the consumer would like, he could call this configure method as many times as he wish...

Anyway, I have just added the next security piece of code as good practice:

```
if (model == newModel)
    return;

...
```